### PR TITLE
statuses poison and lifeleech

### PR DESCRIPTION
**Pull request by JaskRendix**
PR addresses poison and lifeleech:
- now monster can get poisoned on both sides;
- now monster can get "lifeleeched" on both sides too;
- deleted status.py (unused and "odd" because of status status_xxx);
- inserted the apply_status inside poison.py, lifeleech,py and recover.py
- recover works, but the implementation will have a separate PR;
- added poison and lifeleech to the moves below (separate PR);
- added exception handle (fester and blood_bond);

Formula **self.move.potency >= random.random()**

Poison:
```
Move			    Potency		   10 random battles
Battery Acid		-> "potency": 0.35	-> 6 out of 10 times
Fester			-> "potency": 0.8	-> 6 out of 10 times -> it gives poisoning to the user too
Frostbite		-> "potency": 0.2	-> 4 out of 10 times
Rust Bomb		-> "potency": 0.5	-> 7 out of 10 times
Salamander		-> "potency": 0.8	-> 8 out of 10 times
Spray			-> "potency": 0.2	-> 1 out of 10 times
Sting			-> "potency": 0.2	-> 3 out of 10 times
Rot			-> "potency": 1		-> always = 100% probability getting poisoned.
Stone Rot		-> "potency": 1		-> always = 100% probability getting poisoned.
Suck Poison		-> "potency": 1		-> always = 100% probability getting poisoned
Venomous Tentacle	-> "potency": 1		-> always = 100% probability getting poisoned.
```
Lifeleech
```
Blood Bond	-> "potency": 1		-> always -> it gives lifeleeching to the user too
Probiscus	-> "potency": 0.5	-> 6 out of 10 times
Tonguespear	-> "potency": 1		-> always
```
Recover
```
All In		-> "potency": 0.75
Blossom		-> "potency": 1
Fluff Up	-> "potency": 0.5
Font		-> "potency": 1
Refresh		-> "potency": 1
Sudden Glow	-> "potency": 1
```
Source: https://sanglorian.github.io/downloads/Tuxemon/techniques.xlsx

*****

Remember:  open ticket
**reset_status_icons** in combat.py
```
        # add status icons
        for monster in self.active_monsters:
            for status in monster.status:
                if status.icon:
                    # get the rect of the monster
                    rect = self._monster_sprite_map[monster].rect
                    # load the sprite and add it to the display
                    self.load_sprite(
                        status.icon,
                        layer=200,
                        center=rect.topleft, --> icon user visible / icon target off screen
                    )
```
If the user or target faints or get swapped, then the icon remains. Insert the remove.

Remember: open ticket
message (use_success / use_failure) status
-> delete or fix this (already broken):
```
                for status in result.get("statuses", []):
                    m = T.format(
                        status.use_item,
                        {
                            "name": technique.name,
                            "user": status.link.name if status.link else "",
                            "target": status.carrier.name,
                        },
                    )
                    message += "\n" + m
```
-> or upgrade this:
```
            if result["success"]:
                self.suppress_phase_change()
                self.alert(
                    T.format(
                        "combat_status_damage",
                        {"name": target.name, "status": technique.name},
                    )
                )
```

### CODE REVIEW

**Comment by JaskRendix** [comment]
Lifeleech works now:
```
        damage = formula.simple_lifeleech(self.move, self.move.link, target)
        target.current_hp -= damage
        self.move.link.current_hp += damage
```
it needed self.move.link, so the target loses and the user gains hp.
I'll test to see if self.move.carrier works for the target too.

It remains (like the other PR), the problem related to the repetition of the animation and string.
```
Aardorn uses Tonguespear -> target got lifeleech
Target uses XXX
Aardorn gains, Target loses hp
[...]
Aardorn uses Tonguespear
Target uses XXX
Aardorn gains, Target loses hp
Aardorn gains, Target loses hp --> it gains twice and the victim loses twice
```

**Comment by JaskRendix** [comment]
**Repetition of strings and animations**
```
Aardorn uses Tonguespear -> target got lifeleech
Target uses XXX
Aardorn gains, Target loses hp
[...]
Aardorn uses Tonguespear
Target uses XXX
Aardorn gains, Target loses hp
Aardorn gains, Target loses hp --> it gains twice and the victim loses twice
(no matter if Tonguespear fails or not, it keeps accumulating, until the user or the target faints)
```
The string repeated is this one:
```
msgid "combat_status_damage"
msgstr "{name} took {status} damage!"
```

File: mods/tuxemon/db/technique/frostbite.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "animation": "lance_ice",
   "category": "physical",
   "effects": [
+    "status status_poison",
```
Comment (kerizane) [inline mods/tuxemon/db/technique/frostbite.json]: Seems odd...

Comment (JaskRendix) [reply mods/tuxemon/db/technique/frostbite.json]: I'm testing with one monster with Frostbite.
Do you want to get rid of status.py and handle the poisoning (everything else included) inside specific py files (like poison.py)?
If yes, then I'm in.

Comment (JaskRendix) [reply mods/tuxemon/db/technique/frostbite.json]: It works better

**Comment by JaskRendix** [comment]
Without status.py and putting the mechanism inside the poison.py, lifeleech.py and recover.py, it works way better than before. There is no more double animation and double message.

No issues with poison and recover.

The only issue now is that, using Aardorn, if we use tonguespear after the target has been lifeleeched, it crashes with the usual message (the one above).

**Comment by kerizane** [comment]
Sorry, I just meant don't include "hacks" to the gameplay that is just for testing purposes. But, I see that this is just a draft s maybe it was fine...

**Comment by JaskRendix** [comment]
Yeah, no problem, right now it's better without status.py and status status_xxx.
There are a bunch of techniques that need "poison" or "lifeleech" among the effects.

File: tuxemon/technique/techeffect.py [hunk 2]
```diff
@@ -143,4 +143,4 @@ def __init__(
         self._done = False
 
     def apply(self, user: Monster, target: Monster) -> TechEffectResult:
-        ...
+        pass
```
Comment (kerizane) [inline tuxemon/technique/techeffect.py:146]: Why did you change this?

**Review by kerizane — APPROVED** [review]

**Review by kerizane — APPROVED** [review]
